### PR TITLE
Metrics by Tweet Type

### LINKED ISSUE #29: Track metrics by tweet type
Add separate metrics for total tweets, non-retweeted (OC) tweets, retweets, and (optional) total tweets weighted by likes.

### PR DESCRIPTION
**Pull request by rcehemann**
Closes #29

Track metrics for each stream ID by `tweet_type`. Metrics fed to the home screen visualization and stats correspond to `tweet_type == "extended_tweet"` until the new visualization is built (#30). There is now one copy of the original metrics map for each tweet type. The metrics data structure is now a nested map as follows:
```elixir
%{"extended_tweet" => %{
      tweets_processed: 100,
      sum_scores: 1234,
      average_score: 12.34,
      .. 
   },
   "retweeted_status" => %{
      .. 
   },
  ..
}
```

The primary keys are ```["extended_tweet", "retweeted_status", "quoted_status", "text"]```

Open for discussion: do we compute metrics weighted by likes, retweets, etc., as well as the "all-tweets" metrics, on the elixir side or javascript side? I will convert back to draft and keep working if we decide to compute these on the elixir side.

`/admin/dashboard` metrics seem to be working as expected without changes.

### CODE REVIEW

File: lib/sen_tweet/bitfeels/metrics.ex [hunk 1]
```diff
@@ -7,12 +7,16 @@ defmodule SenTweet.Bitfeels.Metrics do
   alias SenTweetWeb.MetricChannel
 
   def create_metrics do
-    %{
-      tweets_processed: 0,
-      sum_scores: 0,
-      average_score: 0,
-      histogram: Enum.map(0..10, &[-1 + 2 * &1 / 11, -1 + 2 * (&1 + 1) / 11, 0])
-    }
+    for type <- ["extended_tweet", "retweeted_status", "quoted_status", "text"],
```
Comment (thekeele) [inline lib/sen_tweet/bitfeels/metrics.ex:10]: Excellent choice with for into here. I suggest pulling some of this out to get a nice oneliner.

`for type <- @tweet_types, into: %{}, do: {type, default_metrics()}`

Another option would be `Enum.into/3`

`Enum.into(@tweet_types, %{}, &{&1, default_metrics())`

**Comment by thekeele** [comment]
> do we compute metrics weighted by likes, retweets, etc., as well as the "all-tweets" metrics, on the elixir side or javascript side?

What do you mean here? I don't recall talking about this.

Should we have an _all_ type that's equivalent to what we had before?

**Comment by thekeele** [comment]
And Javascript is never the answer 😅

**Comment by rcehemann** [comment]
Last time we talked about this PR I mentioned wanting to add the ability to compute metrics weighted by RTs and likes. Since I've been AWOL for a bit I think it's best to do this in a fresh PR. Are we good on this one?

**Pull request merged**